keyboards/converter/adb_usb/matrix.c

```diff
@@ -0,0 +1,231 @@
+    to generate valid code otherwise. Making extra_key uint8_t and constructing codes
+    here via codes = extra_key<<8 | 0xFF; would consistently fail to even LOAD
+    extra_key from memory, and leave garbage in the high byte of codes. I tried
+    dozens of code variations and it kept generating broken assembly output. So
+    beware if attempting to make extra_key code more logical and efficient. */
+    static volatile uint16_t extra_key = 0xFFFF;
+    uint16_t codes;
+    uint8_t key0, key1;
+
+    /* tick of last polling */
+    static uint16_t tick_ms;
+
+    codes = extra_key;
+    extra_key = 0xFFFF;
+
+    if ( codes == 0xFFFF )
+    {
+        // polling with 12ms interval
+        if (timer_elapsed(tick_ms) < 12) return 0;
+        tick_ms = timer_read();
+
+        codes = adb_host_kbd_recv(ADB_ADDR_KEYBOARD);
+
+        // Adjustable keybaord media keys
+        if (codes == 0 && has_media_keys &&
+                (codes = adb_host_kbd_recv(ADB_ADDR_APPLIANCE))) {
+            // key1
+            switch (codes & 0x7f ) {
+            case 0x00:  // Mic
+                codes = (codes & ~0x007f) | 0x42;
+                break;
+            case 0x01:  // Mute
+                codes = (codes & ~0x007f) | 0x4a;
+                break;
+            case 0x02:  // Volume down
+                codes = (codes & ~0x007f) | 0x49;
+                break;
+            case 0x03:  // Volume Up
+                codes = (codes & ~0x007f) | 0x48;
+                break;
+            case 0x7F:  // no code
+                break;
+            default:
+                xprintf("ERROR: media key1\n");
+                return 0x11;
+            }
+            // key0
+            switch ((codes >> 8) & 0x7f ) {
+            case 0x00:  // Mic
+                codes = (codes & ~0x7f00) | (0x42 << 8);
+                break;
+            case 0x01:  // Mute
+                codes = (codes & ~0x7f00) | (0x4a << 8);
+                break;
+            case 0x02:  // Volume down
+                codes = (codes & ~0x7f00) | (0x49 << 8);
+                break;
+            case 0x03:  // Volume Up
+                codes = (codes & ~0x7f00) | (0x48 << 8);
+                break;
+            default:
+                xprintf("ERROR: media key0\n");
+                return 0x10;
+            }
+        }
+    }
+    key0 = codes>>8;
+    key1 = codes&0xFF;
+
+    if (debug_matrix && codes) {
+        print("adb_host_kbd_recv: "); phex16(codes); print("\n");
+    }
+
+    if (codes == 0) {                           // no keys
+        return 0;
+    } else if (codes == 0x7F7F) {   // power key press
+        register_key(0x7F);
+    } else if (codes == 0xFFFF) {   // power key release
+        register_key(0xFF);
+    } else if (key0 == 0xFF) {      // error
+        xprintf("adb_host_kbd_recv: ERROR(%d)\n", codes);
+        // something wrong or plug-in
+        matrix_init();
```

Comment:
Also, this isn't needed here, IIRC.  

The keyboard code calls the init and scan, so this is redundant